src/main/c/org_broadinstitute_hellbender_utils_bwa_BwaMemIndex.c

```diff
@@ -62,46 +93,46 @@ Java_org_broadinstitute_hellbender_utils_bwa_BwaMemIndex_getRefContigNames( JNIE
-	int32_t refID; // reference id
-	int32_t pos; // reference starting position (0-based)
-	int32_t NM; // value for NM tag (number of mismatches)
-	int32_t AS; // value for AS tag (alignment score)
-	int32_t XS; // value for bwa-specific XS tag (suboptimal alignment score)
-	int32_t nCigar; // nCigarOps
-	int32_t cigarOp[nCigarOps]; // len<<4 | op
-	int32_t nMDchars; // length of MD tag
-	char    MDtag[(nMDchars+3)&~3]; // value for MD tag (space allocation rounded up to stay on int32_t boundary)
-	int32_t nXAchars; // length of XA tag
-	char    XAtag[(nXAchars+3)&~3]; // value for XA tag (space rounded up to int32_t boundary)
-
-	// these fields are present only if the read is paired and the mate is mapped, i.e., flag&9==1
-	int32_t mateRefID; // mate's reference id
-	int32_t matePos;   // mate's reference starting position (0-based)
-	int32_t tlen;      // inferred template length
+    int32_t flag_mapQ; // flag<<16 | mapQ (the flag value is a SAM-formatted flag)
+
+    // these fields are present only if the read is mapped, i.e., flag&4==0
+    int32_t refID; // reference id
+    int32_t pos; // reference starting position (0-based)
+    int32_t NM; // value for NM tag (number of mismatches)
+    int32_t AS; // value for AS tag (alignment score)
+    int32_t XS; // value for bwa-specific XS tag (suboptimal alignment score)
+    int32_t nCigar; // nCigarOps
+    int32_t cigarOp[nCigarOps]; // len<<4 | op
+    int32_t nMDchars; // length of MD tag
+    char    MDtag[(nMDchars+3)&~3]; // value for MD tag (space allocation rounded up to stay on int32_t boundary)
+    int32_t nXAchars; // length of XA tag
+    char    XAtag[(nXAchars+3)&~3]; // value for XA tag (space rounded up to int32_t boundary)
+
+    // these fields are present only if the read is paired and the mate is mapped, i.e., flag&9==1
+    int32_t mateRefID; // mate's reference id
+    int32_t matePos;   // mate's reference starting position (0-based)
+    int32_t tlen;      // inferred template length
 } Alignment;
 */
 JNIEXPORT jobject JNICALL
 Java_org_broadinstitute_hellbender_utils_bwa_BwaMemIndex_createAlignments(
-				JNIEnv* env, jclass cls, jobject seqsBuf, jlong idxAddr, jobject optsBuf ) {
-	bwaidx_t* pIdx = (bwaidx_t*)idxAddr;
-	mem_opt_t* pOpts = (*env)->GetDirectBufferAddress(env, optsBuf);
-	char* pSeq = (*env)->GetDirectBufferAddress(env, seqsBuf);
-	size_t bufSize = 0;
-	void* bufMem = jnibwa_createAlignments(pIdx, pOpts, pSeq, &bufSize);
-	jobject alnBuf = (*env)->NewDirectByteBuffer(env, bufMem, bufSize);
-	if ( !alnBuf ) free(bufMem);
-	return alnBuf;
+                JNIEnv* env, jclass cls, jobject seqsBuf, jlong idxAddr, jobject optsBuf ) {
+    bwaidx_t* pIdx = (bwaidx_t*)idxAddr;
+    mem_opt_t* pOpts = (*env)->GetDirectBufferAddress(env, optsBuf);
+    char* pSeq = (*env)->GetDirectBufferAddress(env, seqsBuf);
+    size_t bufSize = 0;
+    void* bufMem = jnibwa_createAlignments(pIdx, pOpts, pSeq, &bufSize);
```

Comment:
After looking at your other pull request, in that one you explicitly handle the possibility that GetDirectBufferAddress returns NULL. In light of that I'd suggest checking if pOpts or pSeq are NULL and if so returning NULL before the call to jnibwa_createAlignments.